pkg/api/v1/types.go

```diff
@@ -1932,6 +1932,33 @@ type SecretList struct {
 	Items []Secret `json:"items" description:"items is a list of secret objects; see http://releases.k8s.io/HEAD/docs/user-guide/secrets.md"`
 }
 
+// Type and constants for component type.
+type ComponentType string
+
+// These are the valid conditions for the component type.
+const (
+	ComponentAPIServer         ComponentType = "apiserver"
+	ComponentControllerManager ComponentType = "controller-manager"
+	ComponentScheduler         ComponentType = "scheduler"
+)
+
+// Component describes metadata about a core component.
+type Component struct {
```

Comment:
That's an option. I didn't do that because i don't really know what "spec" means in this context. But it would at least be more consistent with the Nodes api.